### ISSUE TITLE
Registering entity with redirect link is success but failed to ingest

### ISSUE THREAD

**Issue by dehamzah** [backstage/backstage#6532]
I did try to register the entities using this link: 

https://gitlab.com/dehamzah/catalog-model/tree/master/examples/components/artist-lookup-component.yaml

It registered successfully but it failed to ingest. It's an alias URL for https://gitlab.com/dehamzah/catalog-model/-/blob/master/examples/components/artist-lookup-component.yaml

## Expected Behavior

It should reject the registration when running the dry run API and provide an error feedback to the users.

## Current Behavior

It registers successfully

<img width="1626" alt="Screen Shot 2021-07-19 at 17 18 48" src="https://user-images.githubusercontent.com/6959476/126145625-eef5e05e-2f1a-4572-b258-6480d1f2173c.png">

but it failed to ingest, I got the following error on my backstage app

```
[1] 2021-07-19T10:18:49.856Z catalog warn Unable to read url, Error: Incorrect url: https://gitlab.com/dehamzah/catalog-model/tree/master/examples/components/artist-lookup-component.yaml, Error: Wrong GitLab URL type=plugin entity=location:default/generated-94342f19741b651261931fea98c6783aba08ebbd
```

## Possible Solution

- 

## Steps to Reproduce

1. Go to catalog import page http://localhost:3000/catalog-import
2. Register this link https://gitlab.com/dehamzah/catalog-model/tree/master/examples/components/artist-lookup-component.yaml
3. Check the app log

## Context

Someone in my company registers it using this link and they confuse why the entity is not showing in the catalog list.

## Your Environment

- NodeJS Version: v12.22.1
- Operating System and Version: macOS Catalina 10.15.7 (19H1217)
- Browser Information: Chrome Version 91.0.4472.164 (Official Build) (x86_64)

**Comment by timbonicus** [comment]
This is similar in cause to #6301, I think.

**Comment by regicsolutions** [comment]
> This is similar in cause to #6301, I think.

Yep, I think this one is also connected as well: https://github.com/backstage/backstage/issues/6542#issue-948110138

**Comment by benjdlambert** [comment]
@dehamzah so if you try register this does it work? https://gitlab.com/dehamzah/catalog-model/-/blob/master/examples/components/artist-lookup-component.yaml

The `catalog-import` plugin needs some love unfortunately, the feedback that you get in the UI is that you're getting a location added, but there's no entities being produced from it. I have a feeling that if you use the above URL you should see entities.

It would appear that the support for Github in the `catalog-import` plugin is much better than any other provider so maybe you don't get the necessary feedback that you would expect.

**Comment by dehamzah** [comment]
Yes it will work if I use this link: https://gitlab.com/dehamzah/catalog-model/-/blob/master/examples/components/artist-lookup-component.yaml

I also got the same behavior when trying to import a catalog from a branch that has `/` for example: `feat/test1`. This url returning no entity too: https://gitlab.com/dehamzah/catalog-model/-/blob/feat/test1/examples/components/artist-lookup-component.yaml

<img width="776" alt="Screen Shot 2021-07-26 at 15 04 03" src="https://user-images.githubusercontent.com/6959476/126970597-3f500890-60af-4251-b590-efdcfed06083.png">

I did try it from github and it works

<img width="773" alt="Screen Shot 2021-07-26 at 17 22 21" src="https://user-images.githubusercontent.com/6959476/126974381-224e3cd4-619d-43fe-8c01-838287c76e8d.png">
<img width="776" alt="Screen Shot 2021-07-26 at 17 20 41" src="https://user-images.githubusercontent.com/6959476/126975242-d7b7ff89-991f-4cd1-a4a0-5f2174f8df90.png">


Can you give me guidance where the catalog import integration for gitlab code located @benjdlambert ? cmiiw, i only found github here https://github.com/dehamzah/backstage/blob/master/plugins/catalog-import/src/api/CatalogImportClient.ts

**Comment by benjdlambert** [comment]
Sorry, I should clarify, there's no real special handling of gitlab URL's really, so it doesn't know that `/tree/` should be converted to `/blob/` if it's a file etc, which I think there's special handling of in the Github client as it will check if it's a folder etc first.

The slashes in branch names is also something that could be handled better in the catalog import plugin too, but currently it's a known issue for all providers as without calling out to the provider we don't know if the branch name is part of the URL or not. See #2815

**Comment by kuangp** [comment]
I want to also add on here that since catalog import no longer preforms a full dry run, registering an entity with an improperly formed `catalog-info.yaml` succeeds in adding the location but later fails to process (without any feedback to the user).

This is concerning since before the catalog engine refactor, users are given instant feedback if their entity file is invalid. Now when a user attempts to register a new entity, it either appears after a while (at the next refresh interval) or the never at all if there are errors leading to confusion since the processing error is not exposed anywhere in the frontend.   

In a similar solution to address https://github.com/backstage/backstage/issues/6301, it probably makes sense for to support manually triggering the backend to process these newly registered entities immediately.
